### PR TITLE
IR_M4T2 Add project for task

### PR DESCRIPTION
**Pull request by IrynaRamanets**

### CODE REVIEW

File: Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverCreator.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.epam.gmail.test.driver;
+
+import org.openqa.selenium.WebDriver;
+
+public interface WebDriverCreator {
+
+	public WebDriver createWebDriver();
```
Comment (hanna-bildziuh) [inline Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverCreator.java]: useless access modifier. it's public by default for interface

File: Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	private WebDriverSingleton() {
+	}
+
+	public static void setCreator(WebDriverCreator newCreator) {
+		creator = newCreator;
+	}
+	
+	public static WebDriver getWebDriverInstance() {
+		if (driver == null) {
+			if(creator == null) {
```
Comment (hanna-bildziuh) [inline Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java:20]: Make your code more optimal

```
if (creator == null)
{
    creator = new ChromeDriverCreator();
}
driver = new WebDriverDecorator(creator.createWebDriver());
```

File: Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+				driver = new ChromeDriverCreator().createWebDriver();
+			} else {
+				driver = creator.createWebDriver();
+			}
+			driver = new WebDriverDecorator(driver);
+		}
+		return driver;
+	}
+
+	public static void invalidateWebDriverInstance() {
```
Comment (hanna-bildziuh) [inline Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java:29]: Why do you need the method?
If you add the logic then you son't need singleton I suppose. Think about it

File: Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+public class WebDriverSingleton {
+
+	private static WebDriver driver;
+	
+	private static WebDriverCreator creator;
+
+	private WebDriverSingleton() {
+	}
+
+	public static void setCreator(WebDriverCreator newCreator) {
```
Comment (hanna-bildziuh) [inline Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java:14]: I think WebDriverCreator should be parameter of constructor.

File: Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/module/toolbar/MainToolbar.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+	private final String alertXpath = "//div[@role=\"alert\"]";
+
+	public MainToolbar() {
+		PageFactory.initElements(WebDriverSingleton.getWebDriverInstance(), this);
+	}
+
+	public void moveToSpam(String subject) {
+		DriverUtil.waitForElementToBeClickable(By.xpath("//*[text()=\"" + subject + "\"]")).click();
+
```
Comment (hanna-bildziuh) [inline Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/module/toolbar/MainToolbar.java]: useless empty line

File: Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java (hunk 4 above)
Comment (IrynaRamanets) [reply Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java:14]: I case of singleton constructor should be private, isn't it?

File: Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java (hunk 3 above)
Comment (IrynaRamanets) [reply Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java:29]: In my code I recreate driver instance to reopen browser to simplify change of users.

File: Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java (hunk 4 above)
Comment (hanna-bildziuh) [reply Iryna_Ramanets/Module_4_Framework/M4_HW2/src/main/java/com/epam/gmail/test/driver/WebDriverSingleton.java:14]: My mistake. I mean method `getWebDriverInstance(`

**Pull request merged**